file/builder.go

```diff
@@ -768,6 +768,25 @@ func getStripPathBasedOnProtocols(route kong.Route) (*bool, error) {
 	return route.StripPath, nil
 }
 
+func checkRoutePaths300AndAbove(route FRoute) {
+	pathsWarnings := []string{}
+	for _, p := range route.Paths {
+		if strings.HasPrefix(*p, "~/") || !utils.IsPathRegexLike(*p) {
+			continue
+		}
+		pathsWarnings = append(pathsWarnings, *p)
+	}
+	if len(pathsWarnings) > 0 {
+		cprint.UpdatePrintf(
+			"From the config file, for the route '%s', a regex pattern usage was detected\n"+
```

Comment:
In Route '%s', a path with regular expression was detected. In Kong Gateway versions 3.0 and above, all paths that use regular expressions must be prefixed with a `~` character. Without the `~` prefix, regular expression based paths will not be matched and processed correctly. \n\n Please run foo command to upgrade your config: <command here> \n\n Please refer to the following document for further details: URL.